src/view.c

```diff
@@ -417,7 +418,8 @@ view_apply_region_geometry(struct view *view)
 	geo.width -= margin.left + margin.right;
 	geo.height -= margin.top + margin.bottom;
 
-	if (view->w == geo.width && view->h == geo.height) {
+	if ((int)view->pending_move_resize.width == geo.width
```

Comment:
Could we change `pending_move_resize.{width,height}` to int instead? It doesn't seem to make sense to have them as unsigned while everything else is signed. It also doesn't seem to be used all that much (on latest master, didn't check for this PR):
```
$ egrep -R 'pending_move_resize[.->]*(height|width)' src/
src/xwayland.c: view->pending_move_resize.width = geo.width;
src/xwayland.c: view->pending_move_resize.height = geo.height;
src/xwayland.c:         view->w = view->pending_move_resize.width;
src/xwayland.c:         view->h = view->pending_move_resize.height;
src/xdg.c:                              view->pending_move_resize.width - size.width;
src/xdg.c:                              view->pending_move_resize.height - size.height;
src/xdg.c:      view->pending_move_resize.width = geo.width;
src/xdg.c:      view->pending_move_resize.height = geo.height
```

Reply:
`wlr_box.{width,height}` are `int` so we'd get that change for free if we go that route.

Reply:
Right, I'd prefer to not have the casting everywhere but I don't really mind if we first do the `wlr_box` refactoring or switch the struct member to `int` in the meantime.